unstructured/partition/email.py

```diff
@@ -118,6 +118,14 @@ def build_email_metadata(
     metadata_last_modified: Optional[str] = None,
 ) -> ElementMetadata:
     """Creates an ElementMetadata object from the header information in the email."""
+    signature = None
+    payload = msg.get_payload()
+    if isinstance(payload, list):
+        for item in payload:
+            if item.get_content_type().endswith("signature"):
+                signature = item.get_payload()
+                break
```

Comment:
This is just complicated enough that it could be extracted to a function of its own, even right here internal to this function:
```python
    def find_signature(msg: Message) -> Optional[str]:
        payload = msg.get_payload()
        if not isinstance(payload, list):
            return None
        for item in payload:
            if item.get_content_type().endswith("signature"):
                return item.get_payload()
        return None
```

The benefit being:
- it encapsulates this particular step
- it's easy to extract to module-level later if we end up needing it elsewhere.
- Gives a place for a docstring to explain the contract, like when it works, when it doesn't, etc.
- a few other more subtle benefits

Then the reference below becomes `signature=find_signature(msg),`

Not a request for a change, just something I would be likely to do in a case like this.

Reply:
Good suggestion! Pulled that out into its own function.